.github/workflows/pull-request.yml

```
@@ -260,6 +260,85 @@ jobs:
+              echo "Still waiting for production app to start... ${timeout}s remaining"
+            fi
+          done
+
+          if [ $timeout -eq 0 ]; then
+            echo "Timeout waiting for production application to start"
+            exit 1
+          fi
+          echo "Production app started successfully"
+
+      - name: Stop Production App
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2046:warning:2:8: Quote this to prevent word splitting

(shellcheck)